zio-http/src/main/scala/zio/http/Network.scala

```diff
@@ -0,0 +1,31 @@
+package zio.http
+
+import zio.{Task, ULayer, ZIO, ZLayer}
+
+private[http] trait Network {
+  def findOpenPort: Task[Int]
```

Comment:
There is inherently a race condition here, as what is open at one moment in time may not be open the next moment in time. This is why some APIs are allow you to bind to a specific port, or to bind to any port that is available; and then for the latter, you can find out what the port was after the fact.

Reply:
Good catch, I actually wondered about that at some point during development. Resolved on my branch.